tidysic/settings/formatted_string.py

```diff
@@ -1,55 +1,106 @@
 import re
+from abc import ABC, abstractmethod
 
 from tidysic.file.taggable import Taggable
 
 
+class _Unit(ABC):
+
+    @abstractmethod
+    def write(self, taggable: Taggable):
+        pass
+
+
+class _SubstitutableUnit(_Unit):
+
+    def __init__(self, raw_string: str):
+        pattern = r"(\*?)(.*)\{(\w*)(\:.+)?\}(.*)"
+        match = re.fullmatch(pattern, raw_string)
+
+        if match is None:
+            raise ValueError(f"Could not parse formatted string: '{raw_string}'.")
+
+        self.is_required = (match.group(1) == "*")
+        self.text_before = match.group(2)
+        self.tag_name = match.group(3)
+        self.format_spec = match.group(4)
+        self.text_after = match.group(5)
+
+    def write(self, taggable: Taggable):
```

Comment:
I'm surprised we do not enfore typing the return value 🤔 i.e.,

```python
def write(self, taggable: Taggable) -> str:
    ...
```

This could be done with e.g., adding `--disallow_untyped_defs` and `--check_untyped_defs` to the mypy config of the GH Action.

Reply:
Good call, adding it to the PR.